### ISSUE TITLE
The MNIST example doesn't compile

### ISSUE THREAD

**Issue by fbriggs** [tiny-dnn/tiny-dnn#380]
Trying to compile the code here, with the latest clone of the repo,
https://github.com/tiny-dnn/tiny-dnn/tree/master/examples/mnist
I get the following compilation error:
```
error: too many template arguments for class template 'network'
void construct_net(network<mse, adagrad>& nn) {
```
referencing this line:
`    network<mse, adagrad> nn;`

I think the network class now only takes a single template parameter and the optimizer is specified differently (it is passed in to train() or fit()), but the MNIST example wasn't updated to reflect this change... could be mistaken. Thanks!

**Comment by beru** [comment]
I noticed sample code in https://github.com/tiny-dnn/tiny-dnn/tree/master/examples/mnist/readme.md isn't updated yet.

**Comment by nyanp** [comment]
@fbriggs 
Thanks for reporting, it's solved from [v1.0.0a2](https://github.com/tiny-dnn/tiny-dnn/tree/v1.0.0a2).
